### PR TITLE
drivers: mcux: imx rt flexcan support

### PR DESCRIPTION
**Pull request by rtalbott**
Adds flexcan support to the imx rt series.

### CODE REVIEW

File: drivers/can/can_mcux_flexcan.c [hunk 1]
```diff
@@ -14,6 +14,10 @@
 #include <sys/byteorder.h>
 #include <fsl_flexcan.h>
 
+#if defined(CONFIG_NET_SOCKETS_CAN)
+#include "socket_can_generic.h"
+#endif /* CONFIG_NET_SOCKETS_CAN */
```
Comment (MaureenHelm) [inline drivers/can/can_mcux_flexcan.c:19]: How is this change related to imx rt?

File: drivers/can/can_mcux_flexcan.c [hunk 2]
```diff
@@ -587,17 +591,17 @@ static void mcux_flexcan_transfer_callback(CAN_Type *base,
 
 	switch (status) {
 	case kStatus_FLEXCAN_UnHandled:
-		__fallthrough;
+		/* fallthrough */
```
Comment (MaureenHelm) [inline drivers/can/can_mcux_flexcan.c:594]: why are you changing these fallthroughs? is it a rebasing error?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution @rtalbott . 

Could you please enable this on at least one board? Please add `can` to the board supported list, similar to `boards/arm/frdm_k64f/frdm_k64f.yaml`

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
You need to change the commit message to include a body text in addition to the headline.

File: drivers/can/can_mcux_flexcan.c (hunk 2 above)
Comment (rtalbott) [reply drivers/can/can_mcux_flexcan.c]: yes it was.. fixed now

File: drivers/can/can_mcux_flexcan.c (hunk 1 above)
Comment (rtalbott) [reply drivers/can/can_mcux_flexcan.c]: this is part of an existing feature. I moved it back to the socket can area.

**Review by MaureenHelm — APPROVED** [review]

File: dts/arm/nxp/nxp_rt.dtsi [hunk 3]
```diff
@@ -620,6 +620,57 @@
...
+			clocks = <&ccm IMX_CCM_CAN_CLK 0x68 14>;
+			clk-source = <2>;
+			label = "CAN_1";
+			sjw = <1>;
+			prop-seg = <1>;
+			phase-seg1 = <3>;
+			phase-seg2 = <2>;
+			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <0>;
```
Comment (henrikbrixandersen) [inline dts/arm/nxp/nxp_rt.dtsi:638]: Are these needed?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: dts/arm/nxp/nxp_rt.dtsi (hunk 3 above)
Comment (henrikbrixandersen) [reply dts/arm/nxp/nxp_rt.dtsi:638]: Nevermind me, they are.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Can you set a default for `CAN_MAX_FILTER` in `drivers/can/Kconfig.mcux` when running on i.MX?

**Review by henrikbrixandersen — APPROVED** [review]
LGTM

**Review by alexanderwachter — APPROVED** [review]
